pkg/grpc/api.go

```diff
@@ -289,6 +291,9 @@ func NewAPI(
 		}
 	}
 
+	var extendedMetadata sync.Map
```

Comment:
Why sync.Map here? Would a regular map (or better, a key in the struct below) not be enough?

Reply:
The extendedMetadata sync map is a field already present in the api struct and is used to fill the extended_metadata map in the GetMetadataResponse proto. We could add a new field for the version in the api struct but we would still need to add it to extendedMetadata.

Reply:
I think it makes sense to move it to a separate value, since this is a constant. A sync map is most useful for things that change. But because this is a constant value, it perhaps makes more sense to move it to its own space.

But these are my .02 :)

Reply:
I do not think we should change this as part of this PR.

The PUT / GET `/v1.0/metadata` API stores values in a `sync.map` today. Values may indeed change because we cannot foresee how this API is being used @ItalyPaleAle.

I strongly recommend not changing this.